itests/fevm_test.go

```diff
@@ -1,67 +1,64 @@
 package itests
 
 import (
-	"bytes"
 	"context"
 	"encoding/hex"
-	"os"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/require"
-	cbg "github.com/whyrusleeping/cbor-gen"
 
 	"github.com/filecoin-project/go-address"
 	builtintypes "github.com/filecoin-project/go-state-types/builtin"
 	"github.com/filecoin-project/go-state-types/manifest"
 
 	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
 	"github.com/filecoin-project/lotus/itests/kit"
 )
 
-// TestFEVMBasic does a basic fevm contract installation and invocation
-func TestFEVMBasic(t *testing.T) {
-	// TODO the contract installation and invocation can be lifted into utility methods
-	// He who writes the second test, shall do that.
-	kit.QuietMiningLogs()
+//convert a simple byte array into input data which is a left padded 32 byte array
+func inputDataFromArray(input []byte) []byte {
+	inputData := make([]byte, 32)
+	copy(inputData[32-len(input):], input[:])
+	return inputData
+}
 
+//convert a "from" address into input data which is a left padded 32 byte array
+func inputDataFromFrom(t *testing.T, ctx context.Context, client *kit.TestFullNode, from address.Address) []byte {
+	fromId, err := client.StateLookupID(ctx, from, types.EmptyTSK)
+	require.NoError(t, err)
+
+	senderEthAddr, err := ethtypes.EthAddressFromFilecoinAddress(fromId)
```

Comment:
Not sure if the client.StateLookupID is needed, but I guess this works either way in itests.